app/data/backend_helpers.js

```diff
@@ -0,0 +1,87 @@
+
+var fs = require('fs');
+
+exports.verify = function (data, field_names) {
+    for (var i = 0; i < field_names.length; i++) {
+        if (!data[field_names[i]]) {
+            throw exports.error("missing_data",
+                                field_names[i] + " not optional");
+        }
+    }
+
+    return true;
+}
+
+exports.error = function (code, message) {
+    var e = new Error(message);
+    e.code = code;
+    return e;
+};
+
+exports.file_error = function (err) {
+    return exports.error("file_error", JSON.stringify(err.message));
+}
+
+
+/**
+ * Possible signatures:
+ *  src, dst, callback
+ *  src, dst, can_overwrite, callback
+ */
+exports.file_copy = function () {
+
+    var src, dst, callback;
+    var can_overwrite = false;
+
+    if (arguments.length == 3) {
+        src = arguments[0];
+        dst = arguments[1];
+        callback = arguments[2];
+    } else if (arguments.length == 4) {
+        src = arguments[0];
+        dst = arguments[1];
+        callback = arguments[3];
+        can_overwrite = arguments[2]
+    }
+
+    function copy(err) {
+        var is, os;
+ 
+        if (!err && !can_overwrite) {
+            return callback(backhelp.error("file_exists",
+                                           "File " + dst + " exists."));
+        }
+ 
+        fs.stat(src, function (err) {
+            if (err) {
+                return callback(err);
+            }
+
+            is = fs.createReadStream(src);
+            os = fs.createWriteStream(dst);
+            is.on('end', function () { callback(null); });
+            is.on('error', function (e) { callback(e); });
+            is.pipe(os);
+        });
+    }
+    
+    fs.stat(dst, copy);
+};
+
+
+
+exports.validSetName = function (sn) {
```

Comment:
This raises a question: A lot of the set names contain colons ( : ) - so this must be modified to accept that. We need to take a look through the exact list that the API returns to us to see if there's anymore cases